salt/utils/gitfs.py

```diff
@@ -1939,6 +1940,10 @@ def _fetch(self):
             # pruning only available in pygit2 >= 0.26.2
             pass
         try:
+            # Make sure $HOME env variable is set to prevent
```

Comment:
Since we're modifying the environment anyway can we do it higher up in the stack? Maybe when we run `verify_user`?